src/test/regress/expected/drop_partitioned_table.out

```diff
@@ -395,3 +395,117 @@ NOTICE:  issuing ROLLBACK
 DROP SCHEMA drop_partitioned_table CASCADE;
 NOTICE:  drop cascades to 3 other objects
 SET search_path TO public;
+-- dropping the schema should drop the metadata on the workers
+CREATE SCHEMA partitioning_schema;
+SET search_path TO partitioning_schema;
+CREATE TABLE part_table (
+      col timestamp
+  ) PARTITION BY RANGE (col);
+CREATE TABLE part_table_1
+  PARTITION OF part_table
+  FOR VALUES FROM ('2010-01-01') TO ('2015-01-01');
+SELECT create_distributed_table('part_table', 'col');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- show we have pg_dist_partition entries on the workers
+SELECT run_command_on_workers($$SELECT count(*) FROM  pg_dist_partition where exists(select * from pg_class where pg_class.oid=pg_dist_partition.logicalrelid AND relname ILIKE '%part_table%');$$);
+ run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,2)
+ (localhost,57638,t,2)
+(2 rows)
+
+-- show we have pg_dist_object entries on the workers
+SELECT run_command_on_workers($$SELECT count(*) FROM  citus.pg_dist_object as obj where classid = 1259 AND exists(select * from pg_class where pg_class.oid=obj.objid AND relname ILIKE '%part_table%');$$);
```

Comment:
this might have some conflict with @thanodnl's pg_dist_object change, could be good to wait

Reply:
sounds good, I can rebase once we merge that